scripts/lmdb_matrix.py

```diff
@@ -0,0 +1,164 @@
+import lmdb
+import json
+import click
+from dataclasses import dataclass
+
+from contextlib import contextmanager
+import numpy as np
+
+
+@dataclass
+class GenotypeMatrix:
+    """Store the actual Genotype Matrix"""
+    matrix: np.ndarray
+    transpose: np.ndarray
+    metadata: dict
+
+
+@contextmanager
+def save_excursion(stream):
+    """Context manager to restore stream position after use."""
+    position = stream.tell()
+    try:
+        yield
+    finally:
+        stream.seek(position)
+
+
+def count_lines(stream) -> int:
+    """Count the number of lines in the stream from the current
+    position."""
+    count = 0
+    with save_excursion(stream):
+        while stream.readline().strip():
+            count += 1
+    return count
+
+
+def create_database(db_path: str) -> lmdb.Environment:
+    """Create or open an LMDB environment."""
+    return lmdb.open(db_path, map_size=100 * 1024 * 1024, create=True)
+
+
+def genotype_db_put(db: lmdb.Environment, genotype: GenotypeMatrix) -> bool:
+    metadata = json.dumps(genotype.metadata).encode("utf-8")
+    with db.begin(write=True) as txn:
+        txn.put(b"matrix", genotype.matrix.tobytes())
+        txn.put(b"transpose", genotype.transpose.tobytes())
+        txn.put(b"metadata", metadata)
+    return True
+
+
+def read_geno_file(genotype_file: str) -> GenotypeMatrix:
+    """Read a geno file and return a GenotypeMatrix object."""
+    with open(genotype_file, 'r') as stream:
+        # Read file metadata
+        file_metadata = {}
+        while True:
+            line = stream.readline().strip()
+            if not line:
+                break
+            if line.startswith('#'):
+                continue
+            if line.startswith('@'):
+                key, value = line[1:].split(':', 1)
+                file_metadata[key] = value
+            else:
+                stream.seek(stream.tell() - len(line) - 1)
+                break
+
+        # Read header
+        header = stream.readline().strip().split()
+        metadata_columns = ["Chr", "Locus", "cM",
+                            "Mb"] if "Mb" in header else ["Chr", "Locus", "cM"]
```

Comment:
suggestion:  why not use a unified structure for all files with all the provided headers and using a value such as 'N/A' when a header value is missing?